smriprep/workflows/anatomical.py

```diff
@@ -36,54 +36,56 @@
 from nipype.interfaces.ants import N4BiasFieldCorrection
 
 from niworkflows.engine.workflows import LiterateWorkflow as Workflow
+from niworkflows.utils.spaces import SpatialReferences, Reference
 from niworkflows.interfaces.freesurfer import (
     StructuralReference,
     PatchedLTAConvert as LTAConvert,
 )
 from niworkflows.interfaces.header import ValidateImage
 from niworkflows.interfaces.images import TemplateDimensions, Conform
+from niworkflows.interfaces.nibabel import ApplyMask, Binarize
 from niworkflows.interfaces.nitransforms import ConcatenateXFMs
-from niworkflows.interfaces.utility import KeySelect
-from niworkflows.utils.misc import fix_multi_T1w_source_name, add_suffix
+from niworkflows.utils.misc import add_suffix
 from niworkflows.anat.ants import init_brain_extraction_wf, init_n4_only_wf
-from ..utils.bids import get_outputnode_spec
 from ..utils.misc import apply_lut as _apply_bids_lut, fs_isRunning as _fs_isRunning
-from .norm import init_anat_norm_wf
-from .outputs import init_anat_reports_wf, init_anat_derivatives_wf
-from .surfaces import init_surface_recon_wf
+from .fit.registration import init_register_template_wf
+from .outputs import (
+    init_anat_reports_wf,
+    init_ds_template_wf,
+    init_ds_mask_wf,
+    init_ds_dseg_wf,
+    init_ds_tpms_wf,
+    init_ds_template_registration_wf,
+    init_ds_fs_registration_wf,
+    init_anat_second_derivatives_wf,
+)
+from .surfaces import init_surface_derivatives_wf, init_surface_recon_wf, init_refinement_wf
 
 LOGGER = logging.getLogger("nipype.workflow")
 
 
 def init_anat_preproc_wf(
     *,
-    bids_root,
-    freesurfer,
-    hires,
-    longitudinal,
-    t1w,
-    omp_nthreads,
-    output_dir,
-    skull_strip_mode,
-    skull_strip_template,
-    spaces,
-    debug=False,
-    existing_derivatives=None,
+    bids_root: str,
+    output_dir: str,
+    freesurfer: bool,
+    hires: bool,
+    longitudinal: bool,
+    t1w: list,
+    skull_strip_mode: str,
+    skull_strip_template: Reference,
+    spaces: SpatialReferences,
+    precomputed: dict,
+    debug: bool,
+    omp_nthreads: int,
     name="anat_preproc_wf",
-    skull_strip_fixed_seed=False,
+    skull_strip_fixed_seed: bool = False,
```

Comment:
Type annotations grew out of frustration of not knowing what I was working with (particularly Reference and SpatialReference). The consequence of this was that I need to import `Reference` and `SpatialReference` at the module level, which meant that a mocked `templateflow` prevented the loading of spatial references.